chia/_tests/core/mempool/test_mempool_manager.py

```diff
@@ -1937,3 +1939,74 @@ async def get_coin_records(coin_ids: Collection[bytes32]) -> List[CoinRecord]:
             assert result[1] != MempoolInclusionStatus.FAILED
     except ValidationError as e:
         assert e.code == expected
+
+
+@pytest.mark.anyio
+async def test_fill_rate_block_validation(simulator_and_wallet: OldSimulatorsAndWallets) -> None:
+    """
+    This test creates puzzles of varying sizes and generates blocks with
+    various combinations of them, ensuring they pass block validation.
+    """
+
+    async def send_to_mempool(full_node: FullNodeSimulator, spend_bundle: SpendBundle) -> None:
+        res = await full_node.send_transaction(wallet_protocol.SendTransaction(spend_bundle))
+        assert res is not None and ProtocolMessageTypes(res.type) == ProtocolMessageTypes.transaction_ack
+        res_parsed = wallet_protocol.TransactionAck.from_bytes(res.data)
+        assert res_parsed.status == MempoolInclusionStatus.SUCCESS.value
+
+    async def fill_mempool_with_test_sbs(full_node_api: FullNodeSimulator) -> None:
+        # Create puzzles of varying sizes and farm coins for them
```

Comment:
another way to create bundles with varying sizes would be to include a pointless blob of varying sizes in the puzzle (or solution). That way you may be able to hit the edge case of exactly filling the block